Update hardhat commands with encrypted deployer pk (#104)
scaffold-eth/se-2-docs: docs/deploying/deploy-smart-contracts.mdx

You can generate a random account / private key or add your crypto wallet's private key.

> To create a random account and add the `DEPLOYER_PRIVATE_KEY_ENCRYPTED` to the `.env` file, run:

Reviewer: I think we can remove this line completely since user doesn't / shouldn't manually update the `DEPLOYER_PRIVATE_KEY_ENCRYPTED` in `.env` and it will only be touched by CLI

```
yarn generate
```

> You will be prompted to enter a password which will be used to encrypt your private key. Make sure to remember this password as you'll need it for future deployments and account queries.

Reviewer: If we remove https://github.com/scaffold-eth/se-2-docs/pull/104/files#r1879382950 maybe we should mention here a bit that : 

Running this command will automatically add encrypted private key (`DEPLOYER_PRIVATE_KEY_ENCRYPTED`) in `.env` file.

```mdx-code-block
<Tabs groupId="dev-tool">
<TabItem value="hardhat" label="Hardhat" default>
```

> If you prefer to manually set your own private key, you can use the `yarn account:import` command to encrypt and store your private key. You will get prompted to set a encryption password before storing your `DEPLOYER_PRIVATE_KEY_ENCRYPTED` to `packages/hardhat/.env`.

Reviewer: Maybe we write something like this? 

```suggestion
If you prefer to manually set your own private key, you can use the `yarn account:import` command to encrypt and store your private key. You will get prompted to paste your private key and set encryption password for it.
```

please feel free to reword!
